PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java

```diff
@@ -464,26 +473,66 @@ public void openLinkInBrowser(@NotNull String url) {
 
     private static final Key<String> APP_SERVICE_FILE_ID = new Key<>("APP_SERVICE_FILE_ID");
 
+    @SneakyThrows
     public void openAppServiceFile(final AppServiceFile file, Object context) {
-        final Project project = (Project) context;
-        final FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+        final FileEditorManager fileEditorManager = FileEditorManager.getInstance((Project) context);
         final VirtualFile vFile = getOrCreateVirtualFile(file, fileEditorManager);
-        ApplicationManager.getApplication().invokeLater(() -> fileEditorManager.openFile(vFile, true, true), ModalityState.NON_MODAL);
+        final Task.Modal task = new Task.Modal(null, String.format("Opening file %s...", vFile.getName()), true) {
+            @Override
+            public void run(ProgressIndicator indicator) {
+                indicator.setIndeterminate(true);
+                setVirtualFileContent(vFile, file.getContent())
+                    .doOnSubscribe(indicator::start)
+                    .doAfterTerminate(() -> ApplicationManager.getApplication().invokeLater(() -> {
+                        final FileEditor[] editor = fileEditorManager.openFile(vFile, true, true);
```

Comment:
shall we detect file types  which can be opened by intellij, eg:  for jar, it should not be openned?

Reply:
I tend to let intellij to decide if it can open or not.